### PR TITLE
JAMES-4054 Add X-SMIME-Status in SMIMECheckSignature process

### PR DESCRIPTION
**Pull request by hungphan227**
Will update the ticket number later

### CODE REVIEW

**Comment by chibenwa** [comment]
Please let's rely on https://certificate.nikhef.nl/info/smime-manual.html

I also think working from x509 certificates directly would ease configuration a lot!

**Review by chibenwa — COMMENTED** [review]
`keystore2` please share also the associated public and private PEM files to enable working with it using openssl.

File: server/mailet/integration-testing/src/main/java/org/apache/james/mailets/TemporaryJamesServer.java [hunk 1]
```diff
@@ -139,7 +139,8 @@ public static Builder builder() {
         "pop3server.xml",
         "recipientrewritetable.xml",
         "usersrepository.xml",
-        "smime.p12");
+        "smime.p12",
+        "keystore2");
```
Comment (quantranhong1999) [inline server/mailet/integration-testing/src/main/java/org/apache/james/mailets/TemporaryJamesServer.java:143]: Maybe a better name for this keystore (to reflect the SMIME use case)?

**Review by quantranhong1999 — APPROVED** [review]
We need ticket number

**Review by Arsnael — APPROVED** [review]

**Comment by chibenwa** [comment]
Please let me ~30 mins for review

File: server/mailet/integration-testing/src/test/java/org/apache/james/mailets/crypto/SMIMECheckSignatureIntegrationTest.java [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+            .sendMessageWithHeaders(FROM, RECIPIENT,
+                ClassLoaderUtils.getSystemResourceAsString("eml/mail_with_no_signature.eml"));
+
+        testIMAPClient.connect(LOCALHOST_IP, jamesServer.getProbe(ImapGuiceProbe.class).getImapPort())
+            .login(RECIPIENT, PASSWORD)
+            .select(TestIMAPClient.INBOX)
+            .awaitMessage(awaitAtMostOneMinute);
+        assertThat(testIMAPClient.readFirstMessage()).containsSequence("X-SMIME-Status: Not signed");
+    }
+
```
Comment (chibenwa) [inline server/mailet/integration-testing/src/test/java/org/apache/james/mailets/crypto/SMIMECheckSignatureIntegrationTest.java:141]: Can you add a mail for an unknown CA?

**Review by chibenwa — COMMENTED** [review]
Overall good to me so far

File: server/mailet/integration-testing/src/test/java/org/apache/james/mailets/crypto/SMIMECheckSignatureIntegrationTest.java (hunk 2 above)
Comment (chibenwa) [reply server/mailet/integration-testing/src/test/java/org/apache/james/mailets/crypto/SMIMECheckSignatureIntegrationTest.java:141]: In such case we shall report `X-SMIME-Status: Unknown`

Comment (hungphan227) [reply server/mailet/integration-testing/src/test/java/org/apache/james/mailets/crypto/SMIMECheckSignatureIntegrationTest.java:141]: Do you mean the signature is valid but not trusted?

Comment (chibenwa) [reply server/mailet/integration-testing/src/test/java/org/apache/james/mailets/crypto/SMIMECheckSignatureIntegrationTest.java:141]: Nevermid, after re reading RFC-9219 that served as insoiration this do not make sens.

File: server/mailet/integration-testing/src/test/java/org/apache/james/mailets/crypto/SMIMECheckSignatureIntegrationTest.java [hunk 3]
```diff
@@ -0,0 +1,142 @@
...
+    public void setup(@TempDir File temporaryFolder) throws Exception {
+        MailetContainer mailetContainer = MailetContainer.builder()
+            .putProcessor(CommonProcessors.root())
+            .putProcessor(CommonProcessors.error())
+            .putProcessor(ProcessorConfiguration.transport()
+                .addMailet(MailetConfiguration.BCC_STRIPPER)
+                .addMailet(MailetConfiguration.builder()
+                    .mailet(SMIMECheckSignature.class)
+                    .matcher(All.class)
+                    .addProperty("keyStoreFileName", temporaryFolder.toPath().resolve("conf").resolve("smime_cert_keystore").toAbsolutePath().toString())
```
[the remark this answers is not in the document]
Comment (hungphan227) [reply server/mailet/integration-testing/src/test/java/org/apache/james/mailets/crypto/SMIMECheckSignatureIntegrationTest.java:74]: added .md file to show used commands

**Comment by hungphan227** [comment]
I am a bit curious about this case: When I change email address domain in "From" header to make it different than the email address domain in the signature, the test still pass. I am not sure if it is the right behavior or not.

**Review by chibenwa — APPROVED** [review]

**Pull request merged**